### PR TITLE
Speed up testing

### PR DESCRIPTION
**Pull request by miekg**
* make notification run in the background, this recudes the test_readme
time from 18s to 0.10s
* reduce time for zone reload

* TestServeDNSConcurrent remove entirely. This took a whopping 58s for
  ... ? A few minutes staring didn't reveal wth it is actually testing.
  Making values smaller revealed race conditions in the tests. Remove
  entirely.

* Move many interval values to variables so we can reset them to short
  values for the tests.

* test_large_axfr: make the zone smaller. The number used 64K has no
  rational, make it 64/10 to speed up.
* TestProxyThreeWay: use client with shorter timeout

A few random tidbits in other tests.

Total time saved: 177s (almost 3m) - which makes it worthwhile again to
run the test locally:

this branch:

~~~
ok  	github.com/coredns/coredns/test	10.437s
cd plugin; time go t ./...
5,51s user 7,51s system 11,15s elapsed 744%CPU (
~~~

master:

~~~
ok  	github.com/coredns/coredns/test	35.252s
cd plugin; time go t ./...
157,64s user 15,39s system 50,05s elapsed 345%CPU ()
~~~
tests/ -25s
plugins/ -152s

Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

**Comment by miekg** [comment]
Another 10s can be shaved off as we fix https://github.com/coredns/coredns/issues/4238, meaning all tests run 10s which looks like nice ballpark figure to make local dev. fun again

**Comment by miekg** [comment]
let me fix the timing as I took the wrong value from `time`

**Comment by miekg** [comment]
ah flakyness in travis (of course  it's travis)

**Comment by miekg** [comment]
travis is becoming very very slow

**Comment by SuperQ** [comment]
I'm a fan of CircleCI :grin:

**Comment by miekg** [comment]
yes, would also cut down on external deps, circle-ci is already finished, and that does a minikube k8s test!

**Comment by miekg** [comment]
hmm, my initial commit msg had section headers (#) but these where filtered out because that text is semi-markdown and #-sign are comments... damnit

**Comment by miekg** [comment]
if there are no objects I will merge this

**Review by stp-ip — APPROVED** [review]
/lgtm

File: test/reload_test.go [hunk 1]
```diff
@@ -101,8 +100,6 @@ func TestReloadMetricsHealth(t *testing.T) {
 	}
 	defer c1.Stop()
 
-	time.Sleep(100 * time.Millisecond)
```
Comment (chrisohaver) [inline test/reload_test.go:104]: the sleep was added here... https://github.com/coredns/coredns/pull/1715
not obvious what the rationale was.

**Review by chrisohaver — APPROVED** [review]

**Pull request merged**